wip-0008.md

```diff
@@ -0,0 +1,95 @@
+
+The block limits in [WIP-0007][wip-0007] do normally prevent too many requests from being processed at the same time. However, block reorganizations or reversal in the context of superblock consensus can easily cause several data requests to be retried at the same epoch. That is, under an episode of lack of network consensus, data requests can go from any stage (`commit`, `reveal`, `tally` or `final`) back to the initial `published` stage. This will cause all pending data requests to concurrently enter the `commit` stage in the same epoch, leading to the negative side effects explained before.
+
+The [witnet-rust] implementation of the protocol already enforces some limits to the number of data requests that are posted into a single block. However, these restrictions are only applied when producing block candidates, i.e. block proposers will refrain from including data request transactions if there are already more than a certain number of data requests in the `commit`, `reveal` or `tally` stages; but blocks that exceed those limits are still valid.
+
+There is also a hard limit to the number of retrievals that a [witnet-rust] node will perform and commit to during a single epoch. This limit was implemented back in March 2020 by [pull request #1098][pr1098], and can be adjusted by node operators through the `mining.data_request_max_retrievals_per_epoch` setting in the `witnet.toml` configuration file. Again, this limit is only enforced when mining, and existing consensus rules do not prevent blocks from exceeding such limits.
+
+From a protocol standpoint, all of these restrictions are insufficient, as there exist no block validations that restrict data request concurrency, and alternative protocol implementations may choose not to enforce the same non-consensus limits than [witnet-rust] does, i.e. blocks produced by those alternative implementations could still potentially disrupt [witnet-rust] nodes.
+
+Summing up, there is a need for adding new block validation rules that enforce a reasonable limit to data requests concurrency, so as to ensure availability of the network and allow for more consistent and predictable hardware requirements for nodes. 
+
+
+## Proposal / Specification
+
+Following entry into force of this proposal, all implementations of the Witnet protocol, at all subsequent protocol epochs, MUST add up the combined weight of all data requests in the `commit`, `reveal` or `tally` stages (_unsolved weight_); and reject every block candidate containing any further data requests whose combined weight (_extra weight_) added to the _unsolved weight_ is strictly greater than the `max_dr_weight` constant from [WIP-0007][wip-0007].  
```

Comment:
Data requests that are in reveal or tally stage does not affect to the unsolved_weight because in this case, the weight of these transactions can not increase as commit does. So, the current implementation only take into account data request weights in commit stage

Reply:
Thanks for the clarification!!!